app/controllers/posts_controller.rb

```diff
@@ -122,6 +128,16 @@ def stats
   end
 
   def preview
+    @post ||= Post.new(user: current_user)
+    @post.assign_attributes(post_params(false))
+    @post.board ||= Board.find_by_id(3)
+
+    @author_ids = params.fetch(:post, {}).fetch(:unjoined_author_ids, [])
+    @viewer_ids = params.fetch(:post, {}).fetch(:viewer_ids, [])
+    @settings = process_tags(Setting, :post, :setting_ids)
+    @content_warnings = process_tags(ContentWarning, :post, :content_warning_ids)
+    @labels = process_tags(Label, :post, :label_ids)
```

Comment:
If we don't already have them, tests that viewers and authors are persisted through preview and failed save? Also that preview doesn't persist them to the database.

Reply:
There's at least I think tag tests, because they caught the bug when I started moving things around; I'd have to check to see if they're all there?

Reply:
I do not want to delete comments On Principle but note for future selves that I did check and we do have all necessary tests for associations doing the right things on create/update preview/save.